app/lib/core/services/navigation/nav_service.dart

```diff
@@ -9,7 +11,25 @@ class NavigationService {
     return _navigationKey.currentState.pop();
   }
 
-  Future<dynamic> navigateTo(String routeName) {
-    return _navigationKey.currentState.pushNamed(routeName);
+  Future<dynamic> navigateTo(String routeName, {dynamic arguments}) {
+    return Optional(arguments).isPresent()
+        ? _navigationKey.currentState.pushNamed(routeName, arguments: arguments)
+        : _navigationKey.currentState.pushNamed(routeName);
+  }
```

Comment:
Wrapping arguments with curly braces means it's optional already, I'm pretty sure if-else isn't necessary here. This applies to all nav services methods you did.

Reply:
I think I sent you this before:
```
The following NoSuchMethodError was thrown while handling a gesture:
The method 'pushNamed' was called on null.
Receiver: null
Tried calling: pushNamed<Object>("/A", arguments: null)
```
We still need to handle null case, hence why I cast to optional as an easy way to check without forcing people to add `Optional.empty()` when using this function.

Reply:
oh okay, I didn't know the full context of that error message. If this fixes the problem, then LGTM!